### DISCUSSION TITLE
Expose `ETag` header when serving files

### DISCUSSION THREAD

**Discussion by bywo** [directus/directus#6291] [Feature Requests]
I'm using Directus 9 with Gatsby via `gatsby-source-graphql` and `gatsby-source-filesystem`'s `createRemoteFileNode`. `createRemoteFileNode` will skip redownloading a file if the `ETag` matches. However, fetching a file via Directus resource `/assets/:fileId` doesn't return a `ETag` header. Doing so will significantly improve build times.

**Comment by rijkvanzanten** [comment]
This ties in directly with https://github.com/directus/directus/discussions/5162 (generating a hash checksum on file upload). Once we have 5162, we can easily return that in the ETag header :)

**Comment by jaads** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
